### ISSUE TITLE
Doubtful result on VisDrone

### ISSUE THREAD

**Issue by Disciple7** [ChenhongyiYang/QueryDet-PyTorch#65]
When the paper shows that QueryDet can reach about 28.32 mAP and 48.14 AP50 on VisDrone2018, I only get about 16.0 mAP and 29.8 AP50. Indeed, I did not use the default config the author use (8 GPU, lr=0.01). But it's still too low than it should be. 

Then, I found that there were more questions than I think. 

First, I found that the mAP and AP50 of RetinaNet in this paper (26.21, 44.90), was far more higher than in other papers. In [_VisDrone-DET2019: The Vision Meets Drone Object Detection in Image Challenge Results_](https://openaccess.thecvf.com/content_ICCVW_2019/papers/VISDrone/Du_VisDrone-DET2019_The_Vision_Meets_Drone_Object_Detection_in_Image_Challenge_ICCVW_2019_paper.pdf) , RetinaNet is the worst baseline on VisDrone2019 testdev, which only gets 11.81 mAP and 21.37 AP50 (VisDrone2019 and VisDrone2018 use the same images and annotations)

Then, I trained RetinaNet on VisDrone2018 in Baidu AIStudio, for PaddleDetection also offered pretrained model, scripts and configs for trainning. I still got mAP not reaching 20, just more than 15.0. The picture is attached below (the purple line represents the performance of RetinaNet). 
![RetinaNet trained in AIStudio](https://user-images.githubusercontent.com/36580111/218471677-77394695-1036-4aaa-b6b6-3258072def77.png)

Moreover, the training script of visdrone, "train_visdrone.py", import a module which name is wrong, and will throw ImportError the first time I run it in Pycharm. Did the author truly do the experiment successfully?
```
from models.querydet.detector import RetinaNetQueryDets
```
All of this makes me wonder if the data of this paper are true. I hope if could, the author can offer the successfully trained model that presents the performance the paper noticed, and prove that my doubt is wrong. I spent too much time on these questions, and I need to make sure I'm working on something valuable.

**Comment by ChenhongyiYang** [comment]
Can you please share your training log for QueryDet or our RetinaNet?

**Comment by ChenhongyiYang** [comment]
Hello, 

Firstly, we appreciate your interest in our work. Given your doubt about our work's reproductivity, here we provide the pre-trained model, training log, testing results on the validation set and the full dev codebase as follows. Note that the following result is for our RetinaNet baseline and we will update the QueryDet results later because it takes longer to train a QueryDet model. 

- Model: [model_final.pth](https://drive.google.com/file/d/1FfH7xE8V6SM9qGS9waWtqVWxjRVyyiVZ/view?usp=sharing)
- Training log:  [log.txt](https://drive.google.com/file/d/1vvhpc1MrZjtnEdozW3hzAnSkBDcqdJFs/view?usp=sharing)
- Testing results:  [visdrone_eval.json](https://drive.google.com/file/d/1QyARcwj886E_sihDhz9MEgpcMGe2NXPn/view?usp=sharing) and [visdrone_det_txt](https://drive.google.com/drive/folders/1XkvaXPfy1zw8YYO4iX2g4F_ELcnBH3mE?usp=sharing), which are converted using our released [converting tool](https://github.com/ChenhongyiYang/QueryDet-PyTorch/blob/main/visdrone/json_to_txt.py) and can directly be used by the [official evaluation toolkit](https://github.com/VisDrone/VisDrone2018-DET-toolkit).
-  Original dev code: [querydet_dev.zip](https://drive.google.com/file/d/1jb5bj62WI9bXwW_Wc0UkJV5FePPA_XW6/view?usp=sharing), in which you will also find a READEME_DEV.md file that detailly introduces how to run the training and evaluation code. 

Note that we made minor modifications to the environment setting to reproduce the above result, which is because, at this moment, we do not have access to the machines that I used for developing this work two years ago. Specifically:

- This result was reproduced with PyTorch 1.8.1+cu101, Torchvision 0.9.1+cu101, [Detectron2 0.4.0+Torch 1.8+cu10.1](https://github.com/facebookresearch/detectron2/releases).
- We replaced the APEX FP16 training with the PyTorch AMP training.
- Because we do not have access to the Matlab software at this moment, we evaluated our results using the reimplemented [python evaluation code](https://github.com/tjiiv-cprg/visdrone-det-toolkit-python) developed by the TJIIV group. 

Except for the above modifications, all other settings keep the same as in the original paper. Specifically, we preprocessed the VisDrone dataset using our released [data_prepare.py](https://github.com/ChenhongyiYang/QueryDet-PyTorch/blob/main/visdrone/data_prepare.py) script; the training and testing were conducted using 8 NVIDIA 2080Ti GPUs; all other model and training configs were kept the same as we released before.

The benchmark number on the VisDrone validation set is:
```
evaluating object category 1/10...
evaluating object category 2/10...
evaluating object category 3/10...
evaluating object category 4/10...
evaluating object category 5/10...
evaluating object category 6/10...
evaluating object category 7/10...
evaluating object category 8/10...
evaluating object category 9/10...
evaluating object category 10/10...
Evaluation completed. The performance of the detector is presented as follows.
Average Precision  (AP) @[ IoU=0.50:0.95 | maxDets=500 ] = 31.03953742980957%.
Average Precision  (AP) @[ IoU=0.50      | maxDets=500 ] = 53.60769271850586%.
Average Precision  (AP) @[ IoU=0.75      | maxDets=500 ] = 31.391626358032227%.
Average Recall     (AR) @[ IoU=0.50:0.95 | maxDets=  1 ] = 0.5614168047904968%.
Average Recall     (AR) @[ IoU=0.50:0.95 | maxDets= 10 ] = 5.791536808013916%.
Average Recall     (AR) @[ IoU=0.50:0.95 | maxDets=100 ] = 39.48960494995117%.
Average Recall     (AR) @[ IoU=0.50:0.95 | maxDets=500 ] = 42.35298538208008%.
Class 0 AP = 31.18375015258789%
Class 1 AP = 21.447751998901367%
Class 2 AP = 17.440916061401367%
Class 3 AP = 59.4372444152832%
Class 4 AP = 37.1220588684082%
Class 5 AP = 28.051971435546875%
Class 6 AP = 21.185354232788086%
Class 7 AP = 11.061187744140625%
Class 8 AP = 44.73228454589844%
Class 9 AP = 29.019638061523438%
```

The reproduced result is even better than the numbers that we reported in the original paper. We do not have time to investigate if this improvement was a result of environmental change. However, it does verify the reproductivity and credibility of our paper.

Lastly, although we welcome all friendly discussions on our work, we must kindly remind you that it is irresponsible to make allegations about the credibility of others' research works without thorough verification. As you have mentioned before (both in this issue and your emails), you did not even try our code before making the allegation. Please refrain from making similar allegations to others in the future without a detailed investigation.

Thank you,
Chenhongyi

**Comment by ChenhongyiYang** [comment]
The QueryDet [model](https://drive.google.com/file/d/1KNBwwbxd1HHsURTzLVnv4-II6jOvA2HI/view?usp=sharing) and [training log](https://drive.google.com/file/d/1KRjef5mMgjOyiDAIFxok9tRZRsKmx4_K/view?usp=sharing) are updated here. 

The evaluating result is:
```
evaluating object category 1/10...
evaluating object category 2/10...
evaluating object category 3/10...
evaluating object category 4/10...
evaluating object category 5/10...
evaluating object category 6/10...
evaluating object category 7/10...
evaluating object category 8/10...
evaluating object category 9/10...
evaluating object category 10/10...
Evaluation completed. The performance of the detector is presented as follows.
Average Precision  (AP) @[ IoU=0.50:0.95 | maxDets=500 ] = 32.49479675292969%.
Average Precision  (AP) @[ IoU=0.50      | maxDets=500 ] = 56.03929901123047%.
Average Precision  (AP) @[ IoU=0.75      | maxDets=500 ] = 32.679988861083984%.
Average Recall     (AR) @[ IoU=0.50:0.95 | maxDets=  1 ] = 0.5070593357086182%.
Average Recall     (AR) @[ IoU=0.50:0.95 | maxDets= 10 ] = 6.046351432800293%.
Average Recall     (AR) @[ IoU=0.50:0.95 | maxDets=100 ] = 40.489959716796875%.
Average Recall     (AR) @[ IoU=0.50:0.95 | maxDets=500 ] = 44.53293991088867%.
Class 0 AP = 33.32221603393555%
Class 1 AP = 23.06589126586914%
Class 2 AP = 18.042438507080078%
Class 3 AP = 60.535560607910156%
Class 4 AP = 38.02968978881836%
Class 5 AP = 29.32321548461914%
Class 6 AP = 23.502779006958008%
Class 7 AP = 12.19956111907959%
Class 8 AP = 49.346675872802734%
Class 9 AP = 29.873422622680664%
```

Thank you.

**Comment by ChenhongyiYang** [comment]
Hi, we are glad to hear that. We wish you success with your project. :-)

**Comment by simon1624** [comment]
> 

So, why you failed at first?
